validate filter values

cofiem opened the pull request:
See #305

atruskie commented:
@cofiem this looks great - thanks! 

Can you verify the associated documentation is still accurate? https://github.com/QutBioacoustics/baw-server/wiki/API:-Filtering
As far as I can tell it still is.

cofiem commented:
No worries. I added one more commit to make the class checks consistent.

Yes, the documentation is still correct. I updated it to add a line about the types of values expected.
